fastcrypto/src/secp256r1/mod.rs

```diff
@@ -117,15 +111,14 @@ impl VerifyingKey for Secp256r1PublicKey {
 
 impl Verifier<Secp256r1Signature> for Secp256r1PublicKey {
     fn verify(&self, msg: &[u8], signature: &Secp256r1Signature) -> Result<(), signature::Error> {
-        let pk = signature
-            .recover(msg)
-            .map_err(|_| signature::Error::new())?;
-
-        if pk != *self {
+        // We enforce non malleability, eg. that the s value must be low. This is aligned with
+        // the ECDSA implementation in the secp256k1 crate.
+        if signature.sig.s().is_high().into() {
```

Comment:
Thanks, in theory now that we're segwit we don't need it, but it's a good hygiene to have it.